### PR TITLE
fix: limit WebSocket message backlog on reconnection

### PR DESCRIPTION
**Pull request by petermakowski**
## Done
- Updated WebSocket reconnection logic to limit message backlog

## QA steps

- [ ] Connect to the application and verify WebSocket connection is established
- [ ] Simulate a network interruption (e.g., disable/enable network connection)
- [ ] Confirm WebSocket reconnects successfully
- [ ] Check application logs to ensure no message flood occurs on reconnection
- [ ] Perform several actions in quick succession after reconnection to test the buffer

## Fixes

Fixes: https://bugs.launchpad.net/maas/+bug/2070304
https://warthogs.atlassian.net/browse/MAASENG-3433

## Screenshots

N/A

## Notes

This change introduces a `maxEnqueuedMessages` limit to prevent overwhelming the server with a backlog of messages upon WebSocket reconnection. The limit is set to 30, which accommodates the typical 5-25 messages generated on page load plus some buffer for additional user actions.

### CODE REVIEW

**Review by ndv99 — APPROVED** [review]
LGTM, let's wait for Jacopo's review+approval before merging

**Review by r00ta — APPROVED** [review]

**Pull request merged**
